src/AbstractAlgebra.jl

```diff
@@ -604,17 +604,10 @@ export Generic
 #
 ###############################################################################
 
-getindex(R::Ring, s::String) = PolynomialRing(R, s)
+getindex(R::NCRing, s::Union{String,Char}) = PolynomialRing(R, s)
 
-getindex(R::Ring, s::Char) = PolynomialRing(R, s)
-
-getindex(R::NCRing, s::String) = PolynomialRing(R, s)
-
-getindex(R::NCRing, s::Char) = PolynomialRing(R, s)
-
-getindex(R::Tuple{Ring, T}, s::String) where {T} = PolynomialRing(R[1], s)
-
-getindex(R::Tuple{Ring, T}, s::Char) where {T} = PolynomialRing(R[1], s)
```

Comment:
Is it on purpose that no such method is defined for `NCRing`?

Reply:
I don't think it is on purpose. You can add it if you wish.

Reply:
On the other hand, maybe there is the reason. R[x] is the mathematical notation for a polynomial ring, and I'm not sure off the top of my head whether than includes polynomials over a noncommutative ring. Perhaps @fieker happens to know this off the top of his head.

Reply:
`getindex(R::NCRing, s::String) = PolynomialRing(R, s)` is already defined, so `R["x"]` already works for a noncommutative ring `R`.

Reply:
Ok, then go ahead and make that change and the one below and I will merge this.

Reply:
Ok done, I added tests for the non commutative case too.